### PR TITLE
Wrap element

### PR DESCRIPTION
**Pull request by angelozerr**
Wrap element

### CODE REVIEW

**Comment by angelozerr** [comment]
Tests exists now, @datho7561 please review it again.

**Comment by datho7561** [comment]
If I used linked editing, I can add spaces and newlines into the element name

```xml
<elt|></elt>
```

If I type in `<space>hello<enter>`, then I get:

```xml
<elt hello
></elt hello
  >
```

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMParser.java [hunk 1]
```diff
@@ -98,559 +98,583 @@ public DOMDocument parse(TextDocument document, URIResolverExtensionManager reso
 			}
 			if (previousTokenWasEndTagOpen) {
 				previousTokenWasEndTagOpen = false;
-				if (token != TokenType.EndTag) {
+				boolean linkToEmptyStartTag = false;
+				if (token == TokenType.EndTagClose) {
+					// ex : <\>
```
Comment (datho7561) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMParser.java:103]: `</>`

**Review by datho7561 — CHANGES_REQUESTED** [review]
If you are okay with the behaviour of the linked editing as a result of the parser changes, then I think the PR is good. However, it seems a bit buggy to me. I noticed a small typo in some documentation

**Comment by angelozerr** [comment]
> If I used linked editing, I can add spaces and newlines into the element name

I cannot reproduce, could you do a demo please.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMParser.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMParser.java]: fixed

**Comment by datho7561** [comment]
surround-with-weird-linked-editing

**Comment by angelozerr** [comment]
Oh I see what you mean.  In this case you are using snippet and not linked editing range. I'm not  sure that we can control that.

If you wish to test with linked editing range support, do like your demo to generate the start / end tag element without tag and press on Escape to quit the snippet mode. Type some content in the tag and linked editing range will be activated in this case and you should not have the issue of your demo.

**Review by datho7561 — APPROVED** [review]
> In this case you are using snippet and not linked editing range. 

Okay. Sorry it looks like I just misunderstood what was happening. This looks good to me! Thanks, Angelo!

**Pull request merged**